tests/datadriven/test_variants.py

```diff
@@ -349,6 +341,44 @@ def convertPyvcfNumber(number):
             1 + len(self._formats) + len(self._infos) - gtCounter)
         self.assertEqual(len(keyMap), testMetaLength)
 
+    def testGetVariantsCallSets(self):
+        variantSet = self._gaObject
+        variantSetCompoundId = datamodel.VariantSetCompoundId.parse(
+            variantSet.getId())
+        start = 0
+        end = datamodel.PysamDatamodelMixin.vcfMax
+        callSetIds = variantSet.getCallSetIds()
+        someCallSetIds = callSetIds[0:3]
+        for referenceName in self._referenceNames:
+            # passing None as the callSetIds argument should be equivalent
+            # to passing all of the possible callSetIds as an argument
+            noneRecords = variantSet.getVariants(
+                referenceName, start, end, None, None)
+            allRecords = variantSet.getVariants(
+                referenceName, start, end, None, callSetIds)
+            for noneRecord, allRecord in zip(noneRecords, allRecords):
+                for noneCall, allCall in zip(
+                        noneRecord.calls, allRecord.calls):
+                    self.assertEqual(
+                        noneCall.callSetName, allCall.callSetName)
+
+            # passing an empty list as the callSetIds argument should
+            # return no callsets for any variant
+            emptyRecords = variantSet.getVariants(
+                referenceName, start, end, None, [])
+            for record in emptyRecords:
+                self.assertEqual(len(record.calls), 0)
+
+            # passing some callSetIds as the callSetIds argument should
+            # return only those calls
+            someRecords = variantSet.getVariants(
+                referenceName, start, end, None, someCallSetIds)
+            for record in someRecords:
+                for call in record.calls:
+                    compoundId = datamodel.CallSetCompoundId(
```

Comment:
I think we can zip(record.calls, someCallSetIds) and assert the ID of the call is equal to the specified ID. The protocol requires that the calls be returned in the same order that they are specified in, so this is safe.